### ISSUE TITLE
[BUG] exit not workin in next app router

### ISSUE THREAD

**Issue by aldiabdumalik** [framer/motion#2411]
exit not working, is there something wrong with my code?
"use client"
import { AnimatePresence, motion } from 'framer-motion'
import React from 'react'

```
export default function TransitionEffect() {
  return (
    <>
      <AnimatePresence mode='wait'>
        <motion.div
          key={1}
          className='fixed top-0 bottom-0 right-full w-screen h-screen z-[90] bg-primary'
          initial={{x: '100%', width: '100%'}}
          animate={{x: '0%', width: '0%'}}
          exit={{x:['0%', '100%'], width:['0%', '100%']}}
          transition={{duration:0.8, ease:"easeInOut"}}
        />
        <motion.div
          key={2}
          className='fixed top-0 bottom-0 right-full w-screen h-screen z-[80] bg-light'
          initial={{x: '100%', width: '100%'}}
          animate={{x: '0%', width: '0%'}}
          transition={{delay: 0.2, duration:0.8, ease:"easeInOut"}}
        />
        <motion.div
          key={3}
          className='fixed top-0 bottom-0 right-full w-screen h-screen z-[70] bg-dark'
          initial={{x: '100%', width: '100%'}}
          animate={{x: '0%', width: '0%'}}
          transition={{delay: 0.4, duration:0.8, ease:"easeInOut"}}
        />
      </AnimatePresence>
    </>
  )
}

```

**Comment by laurens-mesure** [comment]
What do you expect here? you aren't unmounting a component to trigger the "exit" animation. Your use-case is almost exactly 1-to-1 with the example in the docs: https://www.framer.com/motion/animate-presence/##exit-animations

**Comment by AaronLayton** [comment]
I am getting this as well - I am doing page navigations though and trying to animate the presence across pages

**Comment by theblondealex** [comment]
I have the same, using Nextjs and wrapping the Layout.tsx children in animatePresence to get entry and exit animations for different pages and routes the entry animations work but exit does not

**Comment by AaronLayton** [comment]
@mattgperry not sure why this was closed as there is no explanation or linked PR?

For those coming back to this, my issue was purely down to my code and how I structured it. On paper it looked like it should have worked, I had the `<AnimatePresence mode="popLayout">` in component 1 just rendering the `{children}`. Then I had a custom component 2 that had

**Component 1**
```tsx
<AnimatePresence mode="popLayout">
   {children}
</AnimatePresence>
```

**Component 2**
```tsx
if (!thingIWanted) return null;

return (
    <motion.div ...>...</div>
)
```

Looked like it should have worked, instead I removed the `<AnimatePresence />` from component 1 and introduced it into component 2

**New Component 2**
```tsx
return (
    <AnimatePresence mode="popLayout">
      {isActiveQuestion && ()
    </AnimatePresence>
}
```
